locator/topology.cc

```diff
@@ -25,11 +25,14 @@ thread_local const endpoint_dc_rack endpoint_dc_rack::default_location = {
     .rack = locator::production_snitch_base::default_rack,
 };
 
+thread_local node::idx_type node::_last_idx;
```

Comment:
Couldn't it be local to topology?

Reply:
I think it must be, or it will keep growing.

Reply:
With node_ptr being lw_shared_ptr<node> that wasn't true since node objects were shared and reused across topology changes, allocating new nodes and their indices only when new nodes are added to the cluster.

Making the idx part of the topology instance makes sense when making the node objects life time tightly coupled with the topology instance.

By the way, with topology of raft we are going to keep nodes forever, after they leave the ring.
We may want to keep them in a different bucket like "_dead_nodes" where they'll have no valid index.
Hopefully, only raft topology code will need to be aware of them.